canonical/ds25
# chore: Rename code generator package

## Done

This is rebased onto #80 , so merge that first. 

- Moves the generator to the `packages` folder
- Renames the generator to `@canonical/generator-ds` so it is used with `yo @canonical/ds` instead of `yo @canonical/canonical-ds`
- Marks the old generator as private so that it is no longer published. We could delete it from the repository after this.
- Fixes some invalid vitest code in the old generator

Fixes #76 
Fixes #75 

## QA
1. `bun i`
2. `cd packages/ds-react-core/src/ui`
3. `yo @canonical/ds:component form/input/textarea` -> see that this creates a new component called TextArea with base component code & tests. See that the component does not spread its `props` (FIxing #76 )
4. `yo @canonical/ds:component form/input/slider --withStyles --withStories` -> see that this creates a new component called Slider with base compoment code, tests, stories, & CSS
5. Try the previous commands but with the old generator: `yo @canonical/canonical-ds`
6. See that it has the same results, except that the old generator spreads component props by default whereas the new component does not. Also, the old generator should log a deprecation message informing users to download the renamed generator.

### PR readiness check

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
